### PR TITLE
Added UIModalPresentationStyle when using legacy Safari Authentication

### PR DESCRIPTION
**Pull request by colin-chan**
### Changes
Legacy authentication using SFSafariViewController can have its modal presentation style changed from its default. For our cases, this allows the view to be presented vertically from the bottom.

Main changes to WebAuth's useLegacyAuthentication method. It accepts a 'withStyle' argument to override the default style. Backward compatible with previous signature by using a default value.

In SafariWebAuth.swift., new private property 'safariPresentationStyle' to store the default or modified enum style.

Example Usage
```
Auth0.webAuth(...)
.useLegacyAuthentication()
.start {...}
```

```
Auth0.webAuth(...)
.useLegacyAuthentication(withStyle: .overFullScreen)
.start {...}
```
### References

Issue raised: https://github.com/auth0/Auth0.swift/issues/274

### Testing

Added new unit test in Auth0Tests/WebAuthSpec.swift

* [x] This change adds unit test coverage
* [ ] This change has been tested on the latest version of the platform/language or why not

### Checklist

* [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [x] All existing and new tests complete without errors
* [x] All active GitHub checks have passed

### CODE REVIEW

**Comment by cocojoe** [comment]
Please can you rebase against master. Thanks

**Comment by colin-chan** [comment]
I have merged in the master changes to my branch.

File: Auth0/WebAuth.swift [hunk 1]
```diff
@@ -190,6 +190,14 @@ public protocol WebAuth: Trackable, Loggable {
     @available(iOS 11, *)
     func useLegacyAuthentication() -> Self
 
+    /**
+     Change modal presentation style if using `SFSafariViewController`.
+     
+     - returns: the same WebAuth instance to allow method chaining
+     */
+    @available(iOS 11, *)
+    func legacyAuthenticationPresentationStyle(_ style: UIModalPresentationStyle) -> Self
```
Comment (cocojoe) [inline Auth0/WebAuth.swift]: I'm not convinced that we need a new protocol method, also a UI method doesn't feel right in this protocol.  Perhaps should be combined as a param in the `useLegacyAuthentication(withStyle: ...)` ?

Comment (colin-chan) [reply Auth0/WebAuth.swift]: I have made the change request, have extended useLegacyAuthentication with a withStyle parameter to accept a custom style or use the previous default value.

File: Auth0/SafariWebAuth.swift [hunk 2]
```diff
@@ -101,8 +102,9 @@ class SafariWebAuth: WebAuth {
         return self
     }
 
-    func useLegacyAuthentication() -> Self {
+    func useLegacyAuthentication(withStyle style: UIModalPresentationStyle = .fullScreen) -> Self {
```
Comment (cocojoe) [inline Auth0/SafariWebAuth.swift]: as this param is required and must be specified when used, the default is not required.

Comment (colin-chan) [reply Auth0/SafariWebAuth.swift]: If I remove the default value then previous versions of this method won't compile. I'm happy to remove the default if you want users to update their code to the default value manually.

Comment (colin-chan) [reply Auth0/SafariWebAuth.swift]: Default value has been removed.

Comment (cocojoe) [reply Auth0/SafariWebAuth.swift]: The previous version will work as your Public extension is the code that allows facilitates that.

Comment (cocojoe) [reply Auth0/SafariWebAuth.swift]: Apologies, you're correct, I had to run this in Xcode as my mental compiler wasn't running properly. I had mixed up the use of the `WebAuth` Protocol vs the `SafariWebAuth` object.

The use case I was thinking about was standard login e.g. Which will be fine as `.webAuth` returns the Protocol and the Public Extension maintains the Public API and Backwards compatibility.

```swift
  Auth0.webAuth()
            .logging(enabled: true)
            .useLegacyAuthentication()
            .start(onAuth)
        }
```

In tests the `SafariWebAuth` is used directly which does not have any default. So please restore :)

File: Auth0/WebAuth.swift [hunk 3]
```diff
@@ -243,3 +244,18 @@ public protocol WebAuth: Trackable, Loggable {
      */
     func clearSession(federated: Bool, callback: @escaping (Bool) -> Void)
 }
+
+public extension WebAuth {
+
```
Comment (cocojoe) [inline Auth0/WebAuth.swift]: The CI is not picking this up. However, on this line there is extra whitespace which should be removed.  I would suggest installing SwiftLint. SwiftLint is auto called in this Libraries build script and will pop up any warnings in the IDE. Thanks

Comment (colin-chan) [reply Auth0/WebAuth.swift]: I have reverted the last commit and deleted the excess whitespace.

**Comment by cocojoe** [comment]
Please rebase your branch for latest changes. thanks

**Comment by colin-chan** [comment]
I have merged in the latest changes.

**Comment by colin-chan** [comment]
Thank you for the approval. Can you you merge my changes for the next release.

**Pull request merged**